### ISSUE TITLE
[BUG]: Destruction Derby Arenas Lights go through walls and Fog Effect Doesn't Go Away from Beginning of Race 

### ISSUE THREAD

**Issue by ps1freak26** [PCSX2/pcsx2#5317]
### Describe the Bug

![DDA pcsx2 hw](https://user-images.githubusercontent.com/91406481/148874927-8cc443bf-72a3-470f-bcb3-4bd1fa572392.jpg)
Red fog at the introduction of race and through all of the race. Different races have different colored fog. Lights from powerups and lamps are see through on walls. Disabling Depth Emulation does get rid of fog but turns the sky black on steelworks. Disabling Depth Emulation does not effect the lights they are still see through.

### Reproduction Steps

Start any race but most noticeable in Steelworks. 

### Expected Behavior

![DDA ps2 snap 1](https://user-images.githubusercontent.com/91406481/148874914-71888ce4-5fbf-4142-a146-0d23504a94b5.jpg)
![DDA pcsx2 sw](https://user-images.githubusercontent.com/91406481/148874916-ac35b709-f3a9-47ee-84af-b7700763d4a3.png)

### PCSX2 Revision

PCSX2 Nightly - v1.7.2212

### Operating System

Windows 10 (64bit)

### CPU

i5 4950

### GPU

1660 super

### GS Settings

The settings are at their defaults expect 4x res but 1x res still has the issue expect software mode.

### Emulation Settings

All settings are at default and preset is at safe mode.

### Logs & Dumps

https://drive.google.com/file/d/1Ho9V0GG115m_xUv4RtmhX3Hp0iVUj7ME/view?usp=sharing

**Comment by ps1freak26** [comment]
I've tested other versions of pcsx2 and 1.6.0 with recent nvidia graphics drivers seem to break the ui and stuff from half the screen and pcsx2 v1.7.2010 seem to have none of the issues. Here are snaps from these different build
![DDA pcsx2 1 6 0](https://user-images.githubusercontent.com/91406481/148879210-f7a74039-92a5-4e8c-8472-ee6054f0f5b2.jpg)
![DDA pcsx2 v1 7 2010](https://user-images.githubusercontent.com/91406481/148879217-9c7155aa-6d58-4179-8ef0-283bb290e319.jpg)
s

**Comment by ps1freak26** [comment]
After much testing with more pcsx2 builds I found that v1.7.2125 is the latest build that works as intended and v1.7.2126 has caused issues with DDA but the builds between these versions I can't be able to download there is 8 builds that are not able to be downloaded between these 2 builds I hope this helps to find what broke this game.

**Comment by ps1freak26** [comment]
:gs hardware:

**Comment by ps1freak26** [comment]
@iMineLink Is there a way you can do a crc hack to disable this stuff for DDA or an option in GSDX?

**Comment by refractionpcsx2** [comment]
Why go straight to hacks? Surely worth seeing if it can be fixed first.

**Comment by RedDevilus** [comment]
It's better to have a pnach equivalent for GS or if possible a regular fix.

**Comment by lightningterror** [comment]
Can't access the dump.

**Comment by ps1freak26** [comment]
[Destruction Derby Arenas Fog and See Through Lights Bug.zip](https://github.com/PCSX2/pcsx2/files/7850602/Destruction.Derby.Arenas.Fog.and.See.Through.Lights.Bug.zip)

**Comment by ps1freak26** [comment]
> Can't access the dump.
[Destruction.Derby.Arenas.Fog.and.See.Through.Lights.Bug.zip](https://github.com/PCSX2/pcsx2/files/7850637/Destruction.Derby.Arenas.Fog.and.See.Through.Lights.Bug.zip)

**Comment by ps1freak26** [comment]
> Can't access the dump.

@lightningterror I uploaded a new one without google drive

**Comment by ps1freak26** [comment]
Does the new dump work does it have to be .gs or .xz format?

**Comment by refractionpcsx2** [comment]
either is fine, a single frame dump will make a .xz but that's just a .gs file in a compressed container, easily extracted

**Comment by iMineLink** [comment]
Confirmed the fact that before https://github.com/PCSX2/pcsx2/commit/2e6c5cde297dca51c5c61f70bb2f4142b261e3c7 the game was working by chance.

What happens is that at draw 434 in your dump, the game does an EE write to address 0x01c0 of a 16x16 rectangle.
The upload is most likely a CLUT, given the fact that 0x01c00 is used as CBP for the draw.
Problem: the depth buffer is also set a 0x01c00, where the depth of the current frame is stored up to draw 434.
So a little part in the top left of the depth buffer should now be marked dirty, as the EE "drawn" there the CLUT.
Instead, our texture cache, especially in `GSTextureCache::InvalidateVideoMem`, is unable to map the color format draw to the depth texture format coordinates, so what it does is instead getting rid completely of the depth buffer at 0x01c00.
This obviously makes the further draws wrong, and causes ultimately the red fog and the see though lights.

Before it worked because the old surfaces cached in the surface cache were not cleared, and so the depth buffer was deleted, its surface recycled in the surface cache, a new depth buffer created using the most recent surface from the cache which, obviously, contained the correct data (aside from the top left 16x16 rectangle)!!! But you cannot rely on a generic surface cache to provide you the correct data as it was happening here.

To fix this is not straight forward at all; The logic to compute generic offsets in pixel from the tex in rt PR (https://github.com/PCSX2/pcsx2/pull/3895) is required to mark the correct region of the buffer as dirty; Hopefully then the target update function will upload the correct data to the buffer.

Please find here an hackfix which seems to fix the misbehaviour in your dump, but is gonna most likely cause other issues elsewhere: https://github.com/iMineLink/pcsx2/commit/940c309a166e692a75ff97fb477eeb260bfbd49c#diff-a08fccf34ca3d5af6f2ba46d887e2f60833415531482f5ac08a9f4848b9f8f45R950

**Comment by ps1freak26** [comment]
@iMineLink First thank you for fixing the Burnout games and trying to find a fix for DDA then further improve it's accuracy rather than just a quick fix. Second I am only a tester I don't know how to run source code so I can't manage to change the code in the source code or the build to try the hackfix. Do you know anybody that can help compile the fix into a build or do I need to use visual studio? Also here is a dump that I ran in the gs debugger in software mode that doesn't have the glitches but it seems using opengl hw settings in the last dump messes with software mode.
[DDA in Software Mode.zip](https://github.com/PCSX2/pcsx2/files/7877029/DDA.in.Software.Mode.zip)

**Comment by iMineLink** [comment]
The new GS dump behaves correctly, no glitches.
I opened a PR for testing the changes using less hacky code for the fix.
Still a work in progress solution.
Wait for this https://github.com/PCSX2/pcsx2/pull/5349 to build and then download the related windows artifacts I suppose from here https://pcsx2.github.io/downloads.html it's the suggested way.

**Comment by ps1freak26** [comment]
I found the builds and I was able to run the exe and the lights and fog are fixed. I went to #5349, checks, windows builds, then artifacts, then avx2 32bit build.

**Comment by ps1freak26** [comment]
Isn't this a regression?

**Comment by refractionpcsx2** [comment]
technically? What's your point?

**Comment by ps1freak26** [comment]
So I would have to find what merged to consider it a regression?

**Comment by refractionpcsx2** [comment]
I don't think it really matters considering we know what it is and there's already a PR which deals with it (but needs work as it breaks other things)
